source/FreeRTOS_DHCPv6.c

```diff
@@ -1156,6 +1158,43 @@ static BaseType_t prvDHCPv6_handleOption( uint16_t usOption,
 
     switch( usOption )
     {
+        case DHCPv6_Option_Status_Code:
+           {
+               FreeRTOS_printf( ( "Status code %02x%02x %02x%02x\n",
+                                  *( pxMessage->ucContents + pxMessage->uxIndex ),
+                                  *( pxMessage->ucContents + pxMessage->uxIndex + 1 ),
+                                  *( pxMessage->ucContents + pxMessage->uxIndex + 2 ),
+                                  *( pxMessage->ucContents + pxMessage->uxIndex + 4 ) ) );
+               uint16_t usStatus = usBitConfig_read_16( pxMessage );
+               size_t uxUsed = pxMessage->uxIndex - pxSet->uxStart;
+
+               FreeRTOS_printf( ( "%s %s with status %u\n",
+                                  ( usOption == DHCPv6_Option_NonTemporaryAddress ) ? "Address assignment" : "Prefix Delegation",
+                                  ( usStatus == 0U ) ? "succeeded" : "failed", usStatus ) );
+
+               if( pxSet->uxOptionLength > uxUsed )
+               {
+                   size_t uxRemain = pxSet->uxOptionLength - uxUsed;
+                   uint8_t ucMessage[ 100 ];
+
+                   if( uxRemain > sizeof( ucMessage ) - 1U )
+                   {
+                       uxRemain = sizeof( ucMessage ) - 1U;
+                   }
+
+                   ( void ) xBitConfig_read_uc( pxMessage, ucMessage, uxRemain );
+                   ucMessage[ uxRemain ] = 0;
+                   FreeRTOS_printf( ( "Msg: '%s'\n", ucMessage ) );
+               }
+
+               /* In case FreeRTOS_printf is not defined. */
+               if( usStatus == 2U )
+               {
+                   pxMessage->xHasError = pdTRUE_UNSIGNED;

```

Comment:
If status is 2, means NoAddrsAvail. (RFC 3315 - sec 24.4)
Why don't we just check if it's not 0U (Success)?

Reply:
~~~diff
-    if( usStatus == 2U )
+    if( usStatus != 0U )
     {
+        /* A status of 2 means: NoAddrsAvail. (RFC 3315 - sec 24.4). */
         pxMessage->xHasError = pdTRUE_UNSIGNED;
     }
~~~